storage/remote/codec_test.go

```diff
@@ -229,3 +234,160 @@ func TestMergeLabels(t *testing.T) {
 		testutil.Equals(t, tc.expected, MergeLabels(tc.primary, tc.secondary))
 	}
 }
+
+type TestSample struct {
```

Comment:
```suggestion
type sample struct {
```

Reply:
This is not possible as it collides with this https://github.com/prometheus/prometheus/blob/90fc6be70f5c11aabcc25a86b5e0bd22a15ae832/storage/remote/queue_manager.go#L650
Anyway, I have replaced it to `testSample`